core/primitives/src/block.rs

```diff
@@ -243,8 +244,16 @@ impl Block {
         );
 
         let new_total_supply = prev.total_supply() + minted_amount.unwrap_or(0) - balance_burnt;
-        let now = to_timestamp(Clock::utc());
-        let time = if now <= prev.raw_timestamp() { prev.raw_timestamp() + 1 } else { now };
+        let time = if let Some(timestamp) = timestamp_override {
+            timestamp
+        } else {
+            let now = to_timestamp(Clock::utc());
+            if now <= prev.raw_timestamp() {
```

Comment:
maybe we should still apply this condition, even if someone has provided the 'override timestamp'?

I'd simply replace the old line 246 with 
let now = timestamp_override.unwrap_or_else(to_timestamp(Clock::utc))

Reply:
not a bad idea to still apply the condition, since fast forward should never be going backwards anyways